libpod/runtime_img.go

```diff
@@ -95,6 +98,260 @@ type imageDecomposeStruct struct {
 	transport   string
 }
 
+func (k *KpodImage) assembleFqName() string {
+	return fmt.Sprintf("%s/%s:%s", k.Registry, k.ImageName, k.Tag)
+}
+
+func (k *KpodImage) assembleFqNameTransport() string {
+	return fmt.Sprintf("%s%s/%s:%s", k.Transport, k.Registry, k.ImageName, k.Tag)
+}
+
+//KpodImage describes basic attributes of an image
+type KpodImage struct {
+	Name          string
+	ID            string
+	fqname        string
+	hasImageLocal bool
+	Runtime
```

Comment:
Does this work?

Reply:
yes

Reply:
So Runtime does not need a type?

Reply:
It's embedding a Runtime struct in here - it's a type, not a variable.

Reply:
That said, I would prefer just making this a pointer to a Runtime, not making this struct a Runtime as well